Grid PR

OshidaBCF opened the pull request:
Added grid.

OshidaBCF committed:
Added Grid

OshidaBCF committed:
Merge branch 'TheoBrs:main' into Swann

OshidaBCF committed:
Fixed PR Requests

OshidaBCF committed:
"finished" grid

OshidaBCF committed:
Changes for PR

OshidaBCF committed:
Fix PR part 2

OshidaBCF committed:
Removed most public stuff

OshidaBCF committed:
Removed Unused Package

OshidaBCF committed:
More PR Fix stuff

OshidaBCF committed:
Finished Grid Movement

I don't know yet how to grab materials from assets via code, so for now the two mats are serialied field

OshidaBCF committed:
Added thingy from discord

OshidaBCF committed:
Finished Grid almost

OshidaBCF committed:
Added button to confirm movement

TheoBrs reviewed (changes requested):
- La traînée violette doit s'afficher avant de cliquer sur le bouton de confirmation, et disparaître après avoir cliqué sur le bouton
- Il faudrait que le personnage se déplace le long du chemin violet, même s'il y a aucune animation pour l'instant, pour que ça fonctionne correctement plus tard
- Il faut tester de mettre un obstacle sur une case aussi, pour voir si le pathfinding le contourne bien
- Il faudrait aussi faire un lien entre le nombres de cases à traverser et le nombre de points d'action disponible. Exemple: si l'entité a 5 points d'actions (PA), elle ne pourra parcourir que 5 cases. Si le joueur veut faire un déplacement d'un nombre de cases supérieure à son nombre de PA, faire en sorte que le chemin qui se dessine soit rouge, et l'empêcher de se déplacer

OshidaBCF committed:
Path now display before moving

OshidaBCF committed:
Added prevention to move if path length is bigger than _currentAP

OshidaBCF committed:
Added obstacle detected

Currently the pathfinder will go around obstacles but it won't remove tiles that aren't part of the "shortest path"

Firzus commented on Assets/Scripts/Board/AStar.cs:
```diff
@@ -0,0 +1,110 @@
+
+public class Cell
+{
+    public Coord coord;
```
pas d'attributs en public

OshidaBCF committed:
Added the red Material for invalid path

Firzus commented on Assets/Scripts/Board/CombatGrid.cs:
```diff
@@ -0,0 +1,69 @@
+
+public class CombatGrid : MonoBehaviour
+{
+    [SerializeField] int maxX;
```
rajouter private après [SerializeField] pour exprimer explicitement le niveau de visibilité

OshidaBCF replied:
Inutile, une variable est toujours privé par défault

Firzus commented on Assets/Scripts/Board/Coord.cs:
```diff
@@ -0,0 +1,33 @@
+public struct Coord
+{
+    public readonly int x;
+    public readonly int y;
```
pas d'attributs en public

Firzus commented on Assets/Scripts/Board/Coord.cs:
```diff
@@ -0,0 +1,33 @@
+       
+    public Coord(int x, int y)
+    {
+        this.x = x;
```
this sera inutile après le changement précédent

OshidaBCF replied:
je ne vois pas pourquoi

Firzus commented on Assets/Scripts/Board/GridElement.cs:
```diff
@@ -0,0 +1,40 @@
+
+public class GridElement
+{
+    Coord coord;
```
pas de niveau d'accès ?

Firzus commented on Assets/Scripts/Board/GridElement.cs:
```diff
@@ -0,0 +1,40 @@
+public class GridElement
+{
+    Coord coord;
+    GameObject gameObject;
```
pareil

Firzus commented on Assets/Scripts/Board/GridElement.cs:
```diff
@@ -0,0 +1,40 @@
+{
+    Coord coord;
+    GameObject gameObject;
+    bool walkable;
```
pareil

Firzus commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -1,19 +1,40 @@
-    private Ability _fAbility2;
-    private Ability _fAbility3;
-    private int _pattoBuff = 0;
+    Ability _fAbility1;
```
il manque les niveaux d’accessibilité

Firzus commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -1,19 +1,40 @@
 
     protected override void AbilitiesInitialization()
     {
+        /* Not instanciated
```
code mort

OshidaBCF replied:
Pas mon code, a voir avec Theo

Firzus commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -23,6 +44,7 @@ protected override void AbilitiesInitialization()
+        */
     }
 
     protected override void CastAbility1(Entity target)
```
code mort

Firzus commented on Assets/Scripts/Character/Player.cs:
reformater le script

OshidaBCF committed:
Finished Grid

OshidaBCF committed:
fixed notwalkable texture

OshidaBCF committed:
forgot to remove one debug

OshidaBCF committed:
i forgot the scene again, needed for the new material

TheoBrs commented on Assets/Scripts/Board/CombatGrid.cs:
```diff
@@ -0,0 +1,80 @@
+        return elements;
+    }
+
+    public Material GetDefaultGridMat()
```
Remplace toutes les fonctions qui ont seulement un Return par " public Material NomDeLaFonction() => valeurARetourner; " pour prendre moins de place

TheoBrs reviewed (changes requested):
Trop de code en public encore

OshidaBCF committed:
better Getter/setter

OshidaBCF committed:
Merge branch 'main' into Swann

OshidaBCF committed:
last changes
